core/src/main/java/hudson/model/BuildAuthorizationToken.java

```diff
@@ -82,10 +81,6 @@ public static void checkPermission(Job<?, ?> project, BuildAuthorizationToken to
             return;
         }
 
-        if (req.getAttribute(ApiTokenProperty.class.getName()) instanceof User) {
-            return;
-        }
```

Comment:
This logic was added in 0271fdb8a21ea6747a8b9a381274b45e13ca944b so that you could use `GET` requests with an API token to trigger a build. A year later, b2a98f6bc6924d1fd25f7da583888c2f4f36d83c broke that logic by removing the request property from `jenkins.security.ApiTokenProperty`. (Later, 814d202716a6c61c7d371c6a62755d296fe199a5 added a new property `jenkins.security.BasicHeaderApiTokenAuthenticator` for a similar purpose.)

Since no one seems to be complaining about the loss of this functionality, and we generally prefer to enforce `POST` on requests with side effects (`GET` requests with per-project tokens were grandfathered in), I am simply deleting the dead code to avoid confusion. My initial attempt at a test case in this PR failed because I thought that just passing an API token would be enough, but in fact `GET` does not work. (And there is no Java API to set the per-project token; you would have to configure it via an HTML form.)